ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -142,6 +145,7 @@ public void run() {
 
           final ShowCompactResponse currentCompactions = txnHandler.showCompact(new ShowCompactRequest());
 
+          metadataCache.ifPresent(c -> c.invalidateAll());
```

Comment:
Could you please add a comment on why do we need to invalidate the cache on every iteration and that's needs to be revisited later?